Tests/PSGetUpdateScriptInfo.Tests.ps1

```diff
@@ -127,29 +128,86 @@ Describe "Update Existing Script Info" -tag CI {
+    It "UpdateScriptFileWithAllFields" {
+        Update-ScriptFileInfo -Path $script:ScriptFilePath @ScriptFileInfoProperties
+
+		$ScriptFileInfo = Test-ScriptFileInfo -Path $script:ScriptFilePath
 
-        Update-ScriptFileInfo -Path $ScriptFilePath @ScriptFileInfoProperties
-		$ScriptFileInfo = Test-ScriptFileInfo -Path $ScriptFilePath
 		foreach ($Prop in $ScriptFileInfoProperties.Keys)
 		{
             $ScriptFileInfo.$Prop | Should be $ScriptFileInfoProperties[$Prop]
 		}
     }
+
+    # Purpose: Update a script file info with an invalid PreRelease string
+    #
+    # Action: Update-ScriptFileInfo [path] -Version 3.2.1-alpha+001
+    #
+    # Expected Result: Update-ScriptFileInfo should throw InvalidCharactersInPreReleaseString errorid.
+    #
+    It "UpdateScriptFileWithInvalidPreReleaseString" {
+        $Version = "3.2.1-alpha+001"
+
+        $expectedErrorMessage = "The PreRelease string contains invalid characters. Please ensure that only characters 'a-zA-Z0-9' and possibly hyphen ('-') at the beginning are in the PreRelease string."
+        $expectedFullyQualifiedErrorId = "InvalidCharactersInPreReleaseString,Validate-PreReleaseString"
+
+        $ScriptBlock = {
+            Update-ScriptFileInfo -Path $script:ScriptFilePath -Version $Version
+        }
+
+        $ScriptBlock | Should -Throw $expectedErrorMessage -ErrorId $expectedFullyQualifiedErrorId
+    }
+
+    # Purpose: Update a script file info with an PreRelease string when the version has insufficient parts.
+    #
+    # Action: Update-ScriptFileInfo [path] -Version 3.2-alpha+001
+    #
+    # Expected Result: Update-ScriptFileInfo should throw InsufficientVersionPartsForPreReleaseString errorid.
+    #
+    It "UpdateScriptFileWithPreReleaseStringAndShortVersion" {
+        $Version = "3.2-alpha001"
+
+        $expectedErrorMessage = "Version has less than 3 parts and a PreRelease string is specified. Version must have at least 3 parts for a PreRelease string to be used."
+        $expectedFullyQualifiedErrorId = "InsufficientVersionPartsForPreReleaseStringUsage,Validate-PreReleaseString"
+
+        $ScriptBlock = {
+            Update-ScriptFileInfo -Path $script:ScriptFilePath -Version $Version
+        }
+
+        $ScriptBlock | Should -Throw $expectedErrorMessage -ErrorId $expectedFullyQualifiedErrorId
+    }
+    
+    # Purpose: Update a script file info with a valid PreRelease string and a version with sufficient parts.
+    #
+    # Action: Update-ScriptFileInfo [path] -Version 3.2.1-alpha001
+    #
+    # Expected Result: Update-ScriptFileInfo should successfully update the version field.
+    #
+    It "UpdateScriptFileWithValidPreReleaseAndVersion" {
+        $Version = "3.2.1-alpha001"
+
+        Update-ScriptFileInfo -Path $script:ScriptFilePath -Version $Version
+
+        $newScriptInfo = Test-ScriptFileInfo -Path $script:ScriptFilePath
+
+        $newScriptInfo.Version | Should -Match $Version
+    }

```

Comment:
Consider adding prerelease validation tests for New-ScriptFileInfo cmdlet.
Many customers are manually adding the PSScriptInfo comment to the script files, given that for Test-ScriptFileInfo cmdlet, consider adding validation of manually created script file with prerelease version.